Assets/HoloToolkit/Utilities/Scripts/AtlasReferenceUpdater/Editor/AtlasReferenceUpdater.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace HoloToolkit.Unity
+{
+    [InitializeOnLoad]
+    public class AtlasReferenceUpdater : UnityEditor.AssetModificationProcessor
+    {
+        private static readonly List<AtlasPrefabReference> References = new List<AtlasPrefabReference>();
+
+        static AtlasReferenceUpdater()
+        {
+            FindAtlasPrefabReferences();
+            UpdateAllReferences();
+            PrefabUtility.prefabInstanceUpdated += PrefabInstanceUpdated;
+        }
+
+        private static void FindAtlasPrefabReferences()
+        {
+            References.Clear();
+            foreach (var reference in FindAllAssets<AtlasPrefabReference>())
+            {
+                if (reference.Atlas == null)
+                {
+                    Debug.LogWarning("No sprite atlas referenced: " + reference.name);
+                    continue;
+                }
+                if (reference.Prefabs.Any(o => o == null))
+                {
+                    Debug.LogWarning("One or more prefab references are null: " + reference.name);
+                    continue;
+                }
+                References.Add(reference);
+            }
+        }
+
+        private static void UpdateAllReferences()
+        {
+            foreach (var atlasPrefabReference in References)
+            {
+                UpdateAtlasReferences(atlasPrefabReference);
+            }
+        }
+
+        private static void PrefabInstanceUpdated(GameObject instance)
+        {
+            var prefab = PrefabUtility.GetPrefabParent(instance) as GameObject;
+            foreach (var atlasPrefabReference in References)
+            {
+                if (atlasPrefabReference.Prefabs.Contains(prefab))
+                {
+                    UpdateAtlasReferences(atlasPrefabReference);
+                }
+            }
+        }
+        private static AssetDeleteResult OnWillDeleteAsset(string assetPath, RemoveAssetOptions options)
+        {
+            var atlasReference = AssetDatabase.LoadAssetAtPath<AtlasPrefabReference>(assetPath);
+            if (atlasReference != null)
+            {
+                EditorApplication.delayCall += FindAtlasPrefabReferences;
+                return AssetDeleteResult.DidNotDelete;
+            }
+
+            var deletedSprite = AssetDatabase.LoadAssetAtPath<Sprite>(assetPath);
+            if (deletedSprite != null)
+            {
+                foreach (var reference in References.Where(re => re.Atlas.ContainsSprite(deletedSprite)))
```

Comment:
Is re a reference?

Reply:
yes

Reply:
I think it's okay. But something like `_ref` or `_reference` is okay.  Up to you.